packages/pci-sdk-iframe/src/pages/app/app.service.ts

```diff
@@ -0,0 +1,66 @@
+import apiClient from '../../apiClient';
+import ICardData from '../../types/ICardData';
+
+/**
+ * Try to display PCI SDK Data.
+ *
+ * If the client is pci-complient the server will return the data.
+ * Otherwise we need to use a verified  2FA token to get the data.
+ */
+async function showCardData(cardId: string): Promise<ICardData> {
+	// Try to get card data from the server
+	try {
+		// If data is obtained just display it. We are done
+		const cardData = await apiClient.getCardData(cardId);
+		return cardData;
+	}
+	catch (err) {
+		// Otherwise we request a 2FA code
+		const { verificationId } = await apiClient.request2FACode();
+		// If request went well we init the verify2FA code and we'll fetch card data again with this code attached
+		return verify2FACode(cardId, verificationId, 'Enter the code we sent you (numbers only):');
```

Comment:
What happens when `apiClient.request2FACode` or `apiClient.verify2FACode` fails? Do we need to catch those errors?

Reply:
When this fails an error will be thrown and the whole function will fail and the .catch function will be executed by the hook.

In future versions we might want to throw different errors and react properly